Konscious.Security.Cryptography.Argon2/Argon2Lane.cs

```diff
@@ -18,18 +20,45 @@ public Argon2Lane(int blockCount)
                     throw new ArgumentOutOfRangeException(nameof(index));
                 }
 
-                return new Argon2Memory(_memory, 128 * index);
+                return _memory.Slice(128*index, 128);
             }
         }
 
-        public int BlockCount
+        public int BlockCount { get; }
+
+        private readonly Memory<ulong> _memory;
+    }
+
+    internal static class SpanExtensions
+    {
+        public static void Blit(this Span<ulong> toBlit, ReadOnlySpan<byte> bytes, int destOffset = 0, int srcOffset = 0, int byteLength = -1)
         {
-            get
+            int remainder = 0;
+            int length;
+            if (byteLength < 0)
             {
-                return _memory.Length / 128;
+                length = 128;
             }
-        }
+            else
+            {
+                length = byteLength / 8;
+                remainder = byteLength - (length * 8);
```

Comment:
can we use **byteLength % 8**  ?

Reply:
Yeah I wasn't really sure why it was that way to begin with. probably because they needed it a few times